app/models/ect_at_school_period.rb

```diff
@@ -61,8 +48,13 @@ class ECTAtSchoolPeriod < ApplicationRecord
 
   # Instance methods
 
-  # lead_provider_name
-  delegate :name, to: :lead_provider, prefix: true, allow_nil: true
+  def lead_provider
+    training_service.latest_lead_provider
+  end
+
+  def delivery_partner
+    training_service.latest_delivery_partner
+  end
```

Comment:
Not sure we want to include this here and call it just `lead_provider` and `delivery_partner` instead of `latest_lead_provider` or `current_lead_provider`...

I wouldn't include these methods in the model and call the SC `ECTAtSchoolPeriods::Training.new...` instead wherever the training information of an ECTAtSchoolPeriod is needed throught the codebase.

Reply:
Ah yeah, you're right - I was being a bit lazy. It looked like more needed changing based on the failing tests, but it's definitely better to remove it now before it gets used more widely. Good shout.

All updated 👍🏽